packages/react-accordion/src/components/AccordionHeader/useAccordionHeaderStyles.ts

```diff
@@ -1,120 +1,82 @@
-    }),
-  ],
-  // Large variant
-  [
-    state => state.size === 'large',
-    theme => ({
-      fontSize: theme.global.type.fontSizes.base[400],
-    }),
-  ],
-  // Extra Large variant
-  [
-    state => state.size === 'extra-large',
-    theme => ({
-      fontSize: theme.global.type.fontSizes.base[500],
-    }),
-  ],
-]);
+const useStyles = makeStyles({
+  root: theme => ({
+    color: theme.alias.color.neutral.neutralForeground1,
+    backgroundColor: theme.alias.color.neutral.neutralBackground1,
+    borderRadius: '2px',
+  }),
+  button: {
+    paddingRight: '10px',
+    paddingLeft: '10px',
+    height: '44px',
+    display: 'flex',
+    alignItems: 'center',
+    cursor: 'pointer',
+  },
+  buttonSmall: {
+    height: '32px',
+  },
+  buttonIconEnd: {
+    justifyContent: 'space-between',
+  },
+  expandIcon: {
+    lineHeight: '0',
+  },
+  expandIconStart: {
+    paddingRight: '8px',
+  },
+  expandIconEnd: {
+    paddingLeft: '8px',
+  },
+  children: theme => ({
+    fontSize: theme.global.type.fontSizes.base[300],
+    fontFamily: theme.global.type.fontFamilies.base,
+    textOverflow: 'ellipsis',
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+  }),
+  childrenSmall: theme => ({
+    fontSize: theme.global.type.fontSizes.base[200],
+  }),
+  childrenLarge: theme => ({
+    fontSize: theme.global.type.fontSizes.base[400],
+  }),
+  childrenExtraLarge: theme => ({
+    fontSize: theme.global.type.fontSizes.base[500],
+  }),
+});
 
 /** Applies style classnames to slots */
 export const useAccordionHeaderStyles = (state: AccordionHeaderState) => {
-  const rootClassName = useRootStyles(state);
-  const buttonClassName = useButtonStyles(state);
-  const expandIconClassNames = useExpandIconStyles(state);
-  const childrenClassNames = useChildrenStyles(state);
+  const styles = useStyles();
+
+  state.className = ax(styles.root, state.className);
+
+  state.button.className = ax(
+    styles.button,
+    state.button.className,
+    state.size === 'small' && styles.buttonSmall,
+    state.expandIconPosition === 'end' && styles.buttonIconEnd,
+  );
 
-  state.className = ax(rootClassName, state.className);
-  state.button.className = ax(buttonClassName, state.button.className);
   if (state.expandIcon) {
-    state.expandIcon.className = ax(expandIconClassNames, state.expandIcon.className);
+    state.expandIcon.className = ax(
+      styles.expandIcon,
+      state.expandIcon.className,
+      state.expandIconPosition === 'start' && styles.expandIconStart,
+      state.expandIconPosition === 'end' && styles.expandIconEnd,
+    );
   }
   if (state.children) {
-    state.children.className = ax(childrenClassNames, state.children.className);
+    state.children.className = ax(
+      styles.children,
+      state.children.className,
+      state.size === 'small' && styles.childrenSmall,
+      state.size === 'large' && styles.childrenLarge,
+      state.size === 'extra-large' && styles.childrenExtraLarge,
```

Comment:
```suggestion
      styles.children,
      state.size === 'small' && styles.childrenSmall,
      state.size === 'large' && styles.childrenLarge,
      state.size === 'extra-large' && styles.childrenExtraLarge,
      state.children.className,
```

💡 `state.*.className` should be always last as it contains overrides. I will be nice to add an automated test for this.